### DISCUSSION TITLE
I have an existing REST Api I want to port to Bun and Honojs

### DISCUSSION THREAD

**Discussion by istvano** [honojs/hono#1571] [Q&A]
Hi

I have an existing Rest API using uws and written in vanilla javascript. I would like to "port" it to Bun and Honojs.

Do I need to use Typescript all the way or can I just use the Javascript code and write the new code in Typescript?

Thanks

**Comment by raphaelkieling** [comment]
In theory bun accepts both js and ts code. You can create 3min POC projects to test this, i cannot see too much issues (at least without check your old code)

**Reply by raphaelkieling** [reply]
Using .js and Hono works with bun.

<img width="1472" alt="image" src="https://github.com/honojs/hono/assets/16259523/3b13e664-a309-456c-bcc1-40b53dd40ae2">

**Comment by istvano** [comment]
thank you!
